<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>UnrollCleanDialog</class>
 <widget class="QDialog" name="UnrollCleanDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>420</width>
    <height>252</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="maximumSize">
   <size>
    <width>1346</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Unroll</string>
  </property>
  <layout class="QVBoxLayout">
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Shaft Radius</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <layout class="QHBoxLayout">
        <item>
         <widget class="QFrame" name="angleFrame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QFrame" name="radiusFrame">
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="radiusLabel">
             <property name="toolTip">
              <string>Cylinder (or cone base) radius</string>
             </property>
             <property name="text">
              <string>Radius</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="radiusDoubleSpinBox">
             <property name="toolTip">
              <string>Shaft radius</string>
             </property>
             <property name="decimals">
              <number>6</number>
             </property>
             <property name="maximum">
              <double>1000000000.000000000000000</double>
             </property>
             <property name="value">
              <double>2.500000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QCheckBox" name="checkBoxDistance">
             <property name="toolTip">
              <string>Computes the distance from the radius (Convergence)</string>
             </property>
             <property name="text">
              <string>Measure Distance</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBoxAxisPosition">
     <property name="title">
      <string>Shaft center position</string>
     </property>
     <layout class="QVBoxLayout">
      <item>
       <widget class="QFrame" name="axisFrame">
        <layout class="QHBoxLayout">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label1">
           <property name="text">
            <string notr="true">X</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="doubleSpinBoxAxisX">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="decimals">
            <number>6</number>
           </property>
           <property name="minimum">
            <double>-1000000000.000000000000000</double>
           </property>
           <property name="maximum">
            <double>1000000000.000000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <spacer>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Fixed</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>10</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="label2">
           <property name="text">
            <string notr="true">Y</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="doubleSpinBoxAxisY">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="decimals">
            <number>6</number>
           </property>
           <property name="minimum">
            <double>-1000000000.000000000000000</double>
           </property>
           <property name="maximum">
            <double>1000000000.000000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <spacer>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Fixed</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>0</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QFrame" name="autoCenterFrame">
        <layout class="QHBoxLayout">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <spacer>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>71</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="unrollRangeGroupBox">
     <property name="toolTip">
      <string>Unroll range (can do multiple turns)</string>
     </property>
     <property name="title">
      <string>Clean</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Octree Level</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spinBoxOctreeLevel">
        <property name="toolTip">
         <string>Level of octree for cleaning (target a size of 5-15cm)</string>
        </property>
        <property name="minimum">
         <number>2</number>
        </property>
        <property name="maximum">
         <number>21</number>
        </property>
        <property name="value">
         <number>9</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="spinBoxOctreeSize">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="toolTip">
         <string>Stop angle (can be negative)</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="suffix">
         <string> m</string>
        </property>
        <property name="decimals">
         <number>3</number>
        </property>
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1000000000000.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>1.000000000000000</double>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="checkBoxClean">
        <property name="toolTip">
         <string>Activate the automatic cleaning</string>
        </property>
        <property name="text">
         <string>Enabled</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>60</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>UnrollCleanDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>209</x>
     <y>231</y>
    </hint>
    <hint type="destinationlabel">
     <x>209</x>
     <y>125</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>UnrollCleanDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>209</x>
     <y>231</y>
    </hint>
    <hint type="destinationlabel">
     <x>209</x>
     <y>125</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
